File: backend/app/models/atm.py
from pydantic import BaseModel


class ContactDetails(BaseModel):
    name: str | None = None
    phone: str | None = None
    email: str | None = None


class ATMDetail(BaseModel):
    atm_id: str
    agency: str | None = None
    category: str | None = None
    close_date: str | None = None
    publish_date: str | None = None
    location: str | None = None
    atm_type: str | None = None
    multi_agency_access: bool | None = None
    panel_arrangement: bool | None = None
    multi_stage: bool | None = None
    description: str | None = None
    other_instructions: str | None = None
    conditions_for_participation: str | None = None
    timeframe_for_delivery: str | None = None
    address_for_lodgement: str | None = None
    addenda_url: str | None = None
    contact_details: ContactDetails | None = None
    document_urls: list[str] = []
    lodgement_url: str | None = None

File: frontend/src/types/atm.ts
export interface ContactDetails {
  name: string | null;
  phone: string | null;
  email: string | null;
}

export interface ATMDetail {
  atm_id: string;
  agency: string | null;
  category: string | null;
  close_date: string | null;
  publish_date: string | null;
  location: string | null;
  atm_type: string | null;
  multi_agency_access: boolean | null;
  panel_arrangement: boolean | null;
  multi_stage: boolean | null;
  description: string | null;
  other_instructions: string | null;
  conditions_for_participation: string | null;
  timeframe_for_delivery: string | null;
  address_for_lodgement: string | null;
  addenda_url: string | null;
  contact_details: ContactDetails | null;
  document_urls: string[];
  lodgement_url: string | null;
}

export interface JobStatus {
  job_id: string;
  status: string;
  steps: string[];
  complete: boolean;
  error: string | null;
}

export interface SearchResult {
  uuid: string;
  title: string;
  href: string;
}
